iotconnect-sdk/nrf-layer-lib/src/iotconnect_mqtt.c

```diff
@@ -438,3 +617,17 @@ void iotc_nrf_mqtt_loop(void) {
         return;
     }
 }
+
+void iotc_nrf_mqtt_disconnect() {
+
+    mqtt_disconnect(&client);
+    //clear list.
+    msg_wait_puback_list_clear();
+}
+
+void iotc_nrf_mqtt_abort() {
```

Comment:
What is the purpose of the abort if disconnect does the same?

Reply:
Hi Nik,

What mqtt_disconnect() does:
1a) send “MQTT Disconnect notification” message to the broker to disconnect cleanly.
1b) close TCP connection.
1c) Fire MQTT event callback with MQTT_EVT_DISCONNECT.

What mqtt_abort() does:
2a) close TCP connection.
2b) Fire MQTT event callback with MQTT_EVT_DISCONNECT.

When LTE link is down, (1a) will not work if we call mqtt_disconnect().